torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4348,19 +4514,31 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            backward_dtypes=floating_types(),
            aten_name='linalg_det',
            sample_inputs_func=sample_inputs_linalg_det,
-           decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack],
+           # Many of the tests fail on ROCm. This is probably
+           # related to the fact that the current linalg.det backward is
+           # unstable if the matrix has repeated singular values, see
+           # https://github.com/pytorch/pytorch/issues/53364
+           decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack, skipCUDAIfRocm],
+           supports_inplace_autograd=False,
+           ),
+    OpInfo('linalg.det',
+           op=torch.linalg.det,
+           variant_test_name='singular',
+           aliases=('det', ),
+           dtypes=double_types(),
+           aten_name='linalg_det',
+           sample_inputs_func=sample_inputs_linalg_det_singular,
+           # Many of the tests fail on ROCm. This is probably
+           # related to the fact that the current linalg.det backward is
+           # unstable if the matrix has repeated singular values, see
+           # https://github.com/pytorch/pytorch/issues/53364
+           decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack, skipCUDAIfRocm],
            supports_inplace_autograd=False,
            skips=(
-               # The following tests fail only on ROCm. This is probably
-               # related to the fact that the current linalg.det backward is
-               # unstable if the matrix has repeated singular values, see
-               # https://github.com/pytorch/pytorch/issues/53364
-               SkipInfo('TestGradients', 'test_fn_grad', device_type='cuda',
-                        dtypes=(torch.float64,), active_if=TEST_WITH_ROCM),
-               SkipInfo('TestGradients', 'test_fn_gradgrad', device_type='cuda',
-                        dtypes=(torch.float64,), active_if=TEST_WITH_ROCM),
-               SkipInfo('TestCommon', 'test_variant_consistency_jit', device_type='cuda',
-                        dtypes=(torch.float64, torch.float32), active_if=TEST_WITH_ROCM),
+               SkipInfo('TestGradients', 'test_fn_gradgrad'),
+               # This test fails because singular inputs cannot be reliably
```

Comment:
@kurtamohler , could you please tell whether you were able to make these `gradgradcheck` tests pass?

Reply:
@nikitaved, `gradgradcheck` failed for the singular case. For nonsingular, it worked